### ISSUE TITLE
documentation: gmres vs minres for symmetric problems

### ISSUE THREAD

**Issue by nschloe** [scipy/scipy#13937]
When going by the book, GMRES and MINRES do the same thing when applied to symmetric matrices. In scipy, they do something different. Can someone explain what's going on here?

MWE:
```python
import scipy.sparse.linalg
import numpy as np


# construct a symmetric matrix
n = 100
data = -np.ones((3, n))
data[1] = 2.0
A = scipy.sparse.spdiags(data, [-1, 0, 1], n, n)

x0 = np.zeros(n)
b = np.ones(n)


def cb(x):
    resnorms.append(np.linalg.norm(A @ x - b, 2))


print()
print("minres():")
resnorms = []
out, info = scipy.sparse.linalg.minres(A, b, x0=x0, maxiter=3, tol=0.0, callback=cb)
print(resnorms)


print()
print("gmres():")
resnorms = []
out, info = scipy.sparse.linalg.gmres(
    A, b, x0=x0, maxiter=3, tol=0.0, callback=cb, callback_type="x"
)
print(resnorms)
```
```
minres():
[9.899494936611665, 9.797958971132712, 9.695359714832659]

gmres():
[10.0, 7.745966692414834, 6.32455532033676, 5.22072926332531]
```

**Comment by tupui** [comment]
Well isn't it linked to what you reported here #13936?

**Comment by nschloe** [comment]
No, it's a different issue. Note how the values don't coincide at all, even if you'd shift them.

**Comment by zhaog6** [comment]
Dear @nschloe @tupui @AtsushiSakai ,
Let me explain it. There are two difference.
1. GMRES has an initial residual output 10.0 in your case but MINRES hasn't it because the `callback` is called in a different location. In `minres.py`, `callback` does not calculate the initial residual norm, so the initial residual is lost.
2. GMRES in SciPy is a restart GMRES method, it means that the method has inner and outer iterations. In fact, for `callback` in SciPy's GMRES, only the residual norm of the outer iteration is returned.

The convergence history of GMRES is as follows. We can see th
![2021-12-01 11-26-52 的屏幕截图](https://user-images.githubusercontent.com/31978442/144169048-e4de7448-7c66-4317-a8f0-8d65f0e6f812.png)
at `callback` does only return the residual norm of the outer iteration
